csrc/codegen.cpp

```diff
@@ -1570,11 +1570,20 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     // At this moment, we only support topk on thread parallelized
     // dimensions. No serial dimension is allowed either.
     ParallelTypeBitmap sorted_parallel_types;
+    IterDomain* batch_id = nullptr;
     for (auto id : sorted_loop_ids) {
-      NVF_ERROR(
-          isParallelTypeThreadDim(id->getParallelType()),
-          "TopK on non-thread dimension is not supported");
-      sorted_parallel_types.set(id->getParallelType());
+      if (isParallelTypeThreadDim(id->getParallelType())) {
+        sorted_parallel_types.set(id->getParallelType());
+      } else if (id->getParallelType() == ParallelType::Group) {
+        NVF_ERROR(
+            batch_id == nullptr,
+            "Multiple batch IDs not supported: ",
+            top->toString());
+        batch_id = id;
+      } else {
+        NVF_THROW(
+            "Invalid parallel type: ", id->toString(), " of ", aop->toString());
```

Comment:
```suggestion
            "Invalid parallel type: ", id->toString(), " of ", top->toString());
```
**[Suggested by AI]** Summary of the fix:

The build was failing because the error-reporting line referenced the variable `aop`, which is not defined (or no longer in scope) at that point in the code.  
The change simply replaces `aop` with the correct, in-scope variable `top` when forming the diagnostic string:

- Before: `"Invalid parallel type: ", id->toString(), " of ", aop->toString()`
- After:  `"Invalid parallel type: ", id->toString(), " of ", top->toString()`

No functional logic is altered; the change just ensures the code compiles and the error message now prints the proper object name.